src/datetime_queries/fetch_entries_from_day_to_hour.rs

```diff
@@ -0,0 +1,146 @@
+    use crate::datetime_queries::utils::FetchEntriesTime;
+    use crate::datetime_queries::{
+        fetch_by_day, fetch_by_hour, fetch_entries_from_day_to_day, fetch_entries_from_day_to_hour,
+        fetch_entries_from_hour_to_day, fetch_entries_from_hour_to_hour,
+    };
+    use crate::retrieval::get_latest_for_entry;
+    use crate::wire_element::WireElement;
+    use ::fixt::prelude::*;
+    use hdk::prelude::*;
+
+    #[test]
+    fn test_fetch_entries_from_day_to_hour() {
+        let mock_day_to_day = fetch_entries_from_day_to_day::MockFetchByDayDay::new();
+        let mock_day_to_hour = fetch_entries_from_day_to_hour::MockFetchByDayHour::new();
+        let mock_hour_to_day = fetch_entries_from_hour_to_day::MockFetchByHourDay::new();
+        let mock_hour_to_hour = fetch_entries_from_hour_to_hour::MockFetchByHourHour::new();
+        let mock_get_latest = get_latest_for_entry::MockGetLatestEntry::new();
+
+        // case 1: fetch by day is called once, and fetch by hour is called 3 times
+        // another case to try out would be same start and end day, so only fetch by hour is called
+        let start_time = FetchEntriesTime {
+            year: 2021,
+            month: 10 as u32,
+            day: 20 as u32,
+            hour: None,
+        };
+        let end_time = FetchEntriesTime {
+            year: 2021,
+            month: 10 as u32,
+            day: 21 as u32,
+            hour: Some(2 as u32),
+        };
+        let base_component = "create".to_string();
+        let wire_element = WireElement::<Example> {
+            header_hash: fixt![HeaderHashB64],
+            entry_hash: fixt![EntryHashB64],
+            entry: Example { number: 1 },
+        };
+        let wire_vec: Vec<WireElement<Example>> = vec![wire_element.clone()];
+        let wire_vec4 = vec![
+            wire_element.clone(),
+            wire_element.clone(),
+            wire_element.clone(),
+            wire_element.clone(),
+        ];
+        let mut mock_by_day = fetch_by_day::MockFetchByDay::new();
+
+        // fetch_entries_by_day should be called for each day in the range
+        mock_by_day
+            .expect_fetch_entries_by_day::<Example>()
+            .with(
+                mockall::predicate::always(),
+                mockall::predicate::always(),
+                mockall::predicate::always(),
+                mockall::predicate::eq(base_component.clone()),
+            )
+            .times(1)
+            .return_const(Ok(wire_vec.clone()));
+
+        let mut mock_by_hour = fetch_by_hour::MockFetchByHour::new();
+        mock_by_hour
+            .expect_fetch_entries_by_hour::<Example>()
+            .with(
+                mockall::predicate::always(),
+                mockall::predicate::eq(start_time.year),
+                mockall::predicate::eq(start_time.month),
+                mockall::predicate::eq(end_time.day),
+                mockall::predicate::always(),
```

Comment:
how come this one is `always`?

Reply:
since the hour changes for each expected function call, im not sure if there is a way to expect multiple calls but different inputs without having to declare the expectation each time?